diff --git a/homework.xcworkspace/xcuserdata/picopeak.xcuserdatad/UserInterfaceState.xcuserstate b/homework.xcworkspace/xcuserdata/picopeak.xcuserdatad/UserInterfaceState.xcuserstate
index 5a8328a..fc7ac9b 100644
Binary files a/homework.xcworkspace/xcuserdata/picopeak.xcuserdatad/UserInterfaceState.xcuserstate and b/homework.xcworkspace/xcuserdata/picopeak.xcuserdatad/UserInterfaceState.xcuserstate differ
diff --git a/homework/Info.plist b/homework/Info.plist
index ac7aff9..f100608 100644
--- a/homework/Info.plist
+++ b/homework/Info.plist
@@ -19,7 +19,7 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>1.4</string>
+	<string>1.5</string>
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleVersion</key>
@@ -41,6 +41,15 @@
 				<key>NSTemporaryExceptionMinimumTLSVersion</key>
 				<string>TLSv1.1</string>
 			</dict>
+			<key>psfshl.pudong-edu.sh.cn</key>
+			<dict>
+				<key>NSIncludesSubdomains</key>
+				<true/>
+				<key>NSTemporaryExceptionAllowsInsecureHTTPLoads</key>
+				<true/>
+				<key>NSTemporaryExceptionMinimumTLSVersion</key>
+				<string>TLSv1.1</string>
+			</dict>
 		</dict>
 	</dict>
 	<key>UILaunchStoryboardName</key>
diff --git a/homework/ViewController.swift b/homework/ViewController.swift
index 8aa953a..7a98b93 100644
--- a/homework/ViewController.swift
+++ b/homework/ViewController.swift
@@ -63,10 +63,15 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
     var db :Connection
     let hwtable :Table
     
+    let fushan_web2 = "www.psfshl.pudong-edu.sh.cn"
+    let fushan_web1 = "www.fushanedu.cn"
+    var fushan_web :String
+    
     required init?(coder aDecoder: NSCoder) {
         let path = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true).first!
         db = try! Connection("\(path)/db.sqlite3.homework")
         hwtable = Table("homework")
+        fushan_web = fushan_web1
         super.init(coder: aDecoder)
     }
 
@@ -74,6 +79,7 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
         let path = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true).first!
         db = try! Connection("\(path)/db.sqlite3.homework")
         hwtable = Table("homework")
+        fushan_web = fushan_web1
         super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
     }
 
@@ -887,9 +893,10 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
     
     /* Main function to get homework */
     func login_and_gethw() {
-        ViewController.obtainViewState("http://www.fushanedu.cn/jxq/jxq_User.aspx") { (vs, error) in
+        ViewController.obtainViewState("http://"+self.fushan_web+"/jxq/jxq_User.aspx") { (vs, error) in
             if (error != nil) {
                 self.alertmsg("请检查网络连接!")
+                self.fushan_web = self.fushan_web2
                 return
             }
             // print("viewstate is ready 1")
@@ -897,7 +904,7 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
             self.login() { (hellomsg, error) in
                 /* try again */
                 print("try again ...")
-                ViewController.obtainViewState("http://www.fushanedu.cn/jxq/jxq_User.aspx") { (vs, error) in
+                ViewController.obtainViewState("http://"+self.fushan_web+"/jxq/jxq_User.aspx") { (vs, error) in
                     if (error != nil) {
                         self.alertmsg("请检查网络连接!")
                         return
@@ -914,7 +921,7 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
                             return
                         }
                         self.isLoggedIn = true
-                        ViewController.obtainViewState("http://www.fushanedu.cn/jxq/jxq_User_jtzyck.aspx") { (vs, error) in
+                        ViewController.obtainViewState("http://"+self.fushan_web+"/jxq/jxq_User_jtzyck.aspx") { (vs, error) in
                             if (error != nil) {
                                 self.alertmsg("请检查网络连接!")
                                 return
@@ -933,7 +940,7 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
     }
 
     func logout_and_login_gethw() {
-        ViewController.obtainViewState("http://www.fushanedu.cn/jxq/jxq_User.aspx") { (vs, error) in
+        ViewController.obtainViewState("http://"+self.fushan_web+"/jxq/jxq_User.aspx") { (vs, error) in
             if (error != nil) {
                 self.alertmsg("请检查网络连接!")
                 return
@@ -952,7 +959,7 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
             print("refreshing data...")
             self.FushanLabel.text = "福外作业 - " + self.name + " ⬇️"
         });
-        ViewController.obtainViewState("http://www.fushanedu.cn/jxq/jxq_User.aspx") { (vs, error) in
+        ViewController.obtainViewState("http://"+self.fushan_web+"/jxq/jxq_User.aspx") { (vs, error) in
             if (error != nil) {
                 self.alertmsg("请检查网络连接!")
                 return
@@ -962,7 +969,7 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
             self.login() { (hellomsg, error) in
                 /* try again */
                 print("try again ...")
-                ViewController.obtainViewState("http://www.fushanedu.cn/jxq/jxq_User.aspx") { (vs, error) in
+                ViewController.obtainViewState("http://"+self.fushan_web+"/jxq/jxq_User.aspx") { (vs, error) in
                     if (error != nil) {
                         self.alertmsg("请检查网络连接!")
                         return
@@ -977,7 +984,7 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
                             return
                         }
                         self.isLoggedIn = true
-                        ViewController.obtainViewState("http://www.fushanedu.cn/jxq/jxq_User_jtzyck.aspx") { (vs, error) in
+                        ViewController.obtainViewState("http://"+self.fushan_web+"/jxq/jxq_User_jtzyck.aspx") { (vs, error) in
                             if (error != nil) {
                                 self.alertmsg("请检查网络连接!")
                                 return
@@ -1061,7 +1068,7 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
 
         let enc: NSStringEncoding = CFStringConvertEncodingToNSStringEncoding(CFStringEncoding(CFStringEncodings.GB_18030_2000.rawValue))
         // let enc :NSStringEncoding = NSUTF8StringEncoding
-        let request = NSMutableURLRequest(URL: NSURL(string: "http://www.fushanedu.cn/jxq/jxq_User.aspx")!, cachePolicy: NSURLRequestCachePolicy.ReloadIgnoringLocalCacheData /* UseProtocolCachePolicy */, timeoutInterval:60.0)
+        let request = NSMutableURLRequest(URL: NSURL(string: "http://"+self.fushan_web+"/jxq/jxq_User.aspx")!, cachePolicy: NSURLRequestCachePolicy.ReloadIgnoringLocalCacheData /* UseProtocolCachePolicy */, timeoutInterval:60.0)
         let paramsLength = postString.lengthOfBytesUsingEncoding(enc)
         let postStringLen = "\(paramsLength)"
         request.addValue("application/x-www-form-urlencoded", forHTTPHeaderField: "Content-Type")
@@ -1130,7 +1137,7 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
         
         let enc: NSStringEncoding = CFStringConvertEncodingToNSStringEncoding(CFStringEncoding(CFStringEncodings.GB_18030_2000.rawValue))
         // let enc :NSStringEncoding = NSUTF8StringEncoding
-        let request = NSMutableURLRequest(URL: NSURL(string: "http://www.fushanedu.cn/jxq/jxq_User.aspx")!, cachePolicy: NSURLRequestCachePolicy.ReloadIgnoringLocalCacheData /* UseProtocolCachePolicy */, timeoutInterval:60.0)
+        let request = NSMutableURLRequest(URL: NSURL(string: "http://"+self.fushan_web+"/jxq/jxq_User.aspx")!, cachePolicy: NSURLRequestCachePolicy.ReloadIgnoringLocalCacheData /* UseProtocolCachePolicy */, timeoutInterval:60.0)
         let paramsLength = postString.lengthOfBytesUsingEncoding(enc)
         let postStringLen = "\(paramsLength)"
         request.addValue("application/x-www-form-urlencoded", forHTTPHeaderField: "Content-Type")
@@ -1201,7 +1208,7 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
         // print("post string is", postString)
         
         let enc: NSStringEncoding = CFStringConvertEncodingToNSStringEncoding(CFStringEncoding(CFStringEncodings.GB_18030_2000.rawValue))
-        let request = NSMutableURLRequest(URL: NSURL(string: "http://www.fushanedu.cn/jxq/jxq_User_jtzyck.aspx")!, cachePolicy: NSURLRequestCachePolicy.ReloadIgnoringLocalCacheData /* UseProtocolCachePolicy */, timeoutInterval:60.0)
+        let request = NSMutableURLRequest(URL: NSURL(string: "http://"+self.fushan_web+"/jxq/jxq_User_jtzyck.aspx")!, cachePolicy: NSURLRequestCachePolicy.ReloadIgnoringLocalCacheData /* UseProtocolCachePolicy */, timeoutInterval:60.0)
         let paramsLength = postString.lengthOfBytesUsingEncoding(enc)
         let postStringLen = "\(paramsLength)"
         request.addValue("application/x-www-form-urlencoded", forHTTPHeaderField: "Content-Type")
@@ -1251,8 +1258,8 @@ class ViewController: UIViewController, UIScrollViewDelegate, LoginViewControlle
             for b in hw_index {
                 let hw_content = doc.xpath("//b[contains(text(),'" + b.text! + "')]/../../following-sibling::tr[1]")
                 var hw_html = hw_content.toHTML!
-                hw_html = hw_html.stringByReplacingOccurrencesOfString("/jxq/UpLoadFolder/", withString: "http://www.fushanedu.cn/jxq/UpLoadFolder/")
-                hw_html = hw_html.stringByReplacingOccurrencesOfString("/WEBADMIN/", withString: "http://www.fushanedu.cn/")
+                hw_html = hw_html.stringByReplacingOccurrencesOfString("/jxq/UpLoadFolder/", withString: "http://"+self.fushan_web+"/jxq/UpLoadFolder/")
+                hw_html = hw_html.stringByReplacingOccurrencesOfString("/WEBADMIN/", withString: "http://"+self.fushan_web+"/")
                 hw.append(b.toHTML! + "<BR><BR>" + hw_html)
             }
             // hw.append("测试今日作业")
